src/middleware.ts

```diff
@@ -215,7 +215,7 @@ export const persist = <S extends State>(
     )
   }
 
-  const setItem = async () => {
+  const setItem = async function () {
```

Comment:
@dai-shi this fixes https://github.com/pmndrs/zustand/pull/343#issuecomment-808225863